src/api/app/controllers/webui/package_controller.rb

```diff
@@ -815,36 +817,10 @@ def set_job_status
   def live_build_log
     required_parameters :arch, :repository
 
-    if Project.exists_by_name(params[:project])
-      @project = Project.get_by_name(params[:project])
-    else
-      flash[:error] = "Couldn't find project '#{params[:project]}'. Are you sure it still exists?"
-      redirect_back(fallback_location: root_path)
-      return
-    end
-
-    begin
-      @package = Package.get_by_project_and_name(@project,
-                                                 params[:package],
-                                                 {use_source:           false,
-                                                  follow_multibuild:    true,
-                                                  follow_project_links: true})
-    rescue Package::UnknownObjectError
-      flash[:error] = "Couldn't find package '#{params[:package]}' in project '#{@project.to_param}'. Are you sure it exists?"
-      redirect_to project_show_path(@project.to_param)
-      return
-    end
-
-    if @package && !@package.check_source_access?
-      flash[:error] = 'Could not access build log'
-      redirect_to action: :show, project: @project.name, package: @package.name
-      return
-    end
-
     @build_container = params[:package] # for remote and multibuild package
-    @package ||= params[:package] # for remote case
-    @arch = params[:arch]
-    @repo = params[:repository]
+    # Make sure objects don't contain invalid chars (eg. '../')
+    @arch = Architecture.find_by(name: params[:arch]).try(:name)
+    @repo = Repository.find_by(name: params[:repository]).try(:name)
```

Comment:
![This is not the Repository you are looking for](https://cloud.githubusercontent.com/assets/514785/20964192/ca039564-bc70-11e6-8b75-72d672315ad5.gif)

This is not the Repository you are looking for.

```
Repository.where(name: 'openSUSE_13.2').count
=> 4397
```